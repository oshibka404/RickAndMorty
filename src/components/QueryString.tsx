import React from 'react';
import {StyleSheet, View, TextInput, ActivityIndicator} from 'react-native';

interface Props {
    query: string;
    setQuery(q: string): void;
    isLoading: boolean;
}

export function QueryString({query, setQuery, isLoading} : Props) {
    return (
        <View style={styles.header}>
            <TextInput
                placeholder='Search character by name'
                returnKeyType='search'
                style={styles.search}
                clearButtonMode='while-editing'
                onChangeText={setQuery}
                value={query}
            />
            {isLoading && <ActivityIndicator />}
        </View>
    )
}

const styles = StyleSheet.create({
    header: {
        flexDirection: 'row',
        borderColor: '#ccc',
        borderBottomWidth: 1,
        paddingHorizontal: 5
    },
    search: {
        height: 40,
        flex: 1
    }
});
