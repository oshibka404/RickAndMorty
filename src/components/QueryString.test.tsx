import React from 'react';
import renderer from 'react-test-renderer';
import { QueryString } from './QueryString';

describe('Query string component', () => {
  test('Renders correctly', () => {
    const tree = renderer.create(
      <QueryString
        isLoading={false}
        query="Betty"
        setQuery={jest.fn()}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
