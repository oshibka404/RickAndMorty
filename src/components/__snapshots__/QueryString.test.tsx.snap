// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query string component Renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 1,
      "borderColor": "#ccc",
      "flexDirection": "row",
      "paddingHorizontal": 5,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    clearButtonMode="while-editing"
    onChangeText={[MockFunction]}
    placeholder="Search character by name"
    rejectResponderTermination={true}
    returnKeyType="search"
    style={
      Object {
        "flex": 1,
        "height": 40,
      }
    }
    underlineColorAndroid="transparent"
    value="Betty"
  />
</View>
`;
